### PR TITLE
add support for changing traffic type in spectrum apps

### PR DESCRIPTION
**Pull request by gposton**
This PR needs merged and vendored in prior to merging this: https://github.com/cloudflare/cloudflare-go/pull/350

### CODE REVIEW

File: vendor/github.com/cloudflare/cloudflare-go/spectrum.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ type SpectrumApplication struct {
 	IPFirewall    bool                          `json:"ip_firewall,omitempty"`
 	ProxyProtocol bool                          `json:"proxy_protocol,omitempty"`
 	TLS           string                        `json:"tls,omitempty"`
+	TrafficType   string                        `json:"traffic_type,omitempty"`
```
Comment (patryk) [inline vendor/github.com/cloudflare/cloudflare-go/spectrum.go]: Please PR this in https://github.com/cloudflare/cloudflare-go/blob/master/spectrum.go
We'll vendor the change back here in separate PR.

File: cloudflare/resource_cloudflare_spectrum_application.go [hunk 2]
```diff
@@ -33,6 +33,15 @@ func resourceCloudflareSpectrumApplication() *schema.Resource {
 				Required: true,
 			},
 
+			"traffic_type": {
+				Type:     schema.TypeString,
+				Default:  "tcp",
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"tcp", "http", "https",
```
Comment (patryk) [inline cloudflare/resource_cloudflare_spectrum_application.go]: Valid values are: `direct`, `http`, `https`
https://api.cloudflare.com/#spectrum-applications-properties

**Comment by patryk** [comment]
Please add new field to docs.

File: vendor/github.com/cloudflare/cloudflare-go/spectrum.go (hunk 1 above)
Comment (gposton) [reply vendor/github.com/cloudflare/cloudflare-go/spectrum.go]: https://github.com/cloudflare/cloudflare-go/pull/350

File: cloudflare/resource_cloudflare_spectrum_application.go (hunk 2 above)
Comment (gposton) [reply cloudflare/resource_cloudflare_spectrum_application.go]: fixed.  thanks!

Comment (gposton) [reply cloudflare/resource_cloudflare_spectrum_application.go]: fixed, thanks!

**Comment by patryk** [comment]
@gposton could you add new field to docs? https://github.com/terraform-providers/terraform-provider-cloudflare/blob/master/website/docs/r/spectrum_application.html.markdown

**Comment by gposton** [comment]
> @gposton could you add new field to docs? https://github.com/terraform-providers/terraform-provider-cloudflare/blob/master/website/docs/r/spectrum_application.html.markdown

Done!

**Comment by patryk** [comment]
Great, thanks @gposton !

**Pull request merged**